pkg/yurthub/otaupdate/ota.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2022 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package otaupdate
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+// TODO(hxc): should use pkg/controller/daemonpodupdater.PodNeedUpgrade
+const (
+	PodNeedUpgrade corev1.PodConditionType = "PodNeedUpgrade"
+)
+
+// GetPods return pod list
+func GetPods(clientset kubernetes.Interface, nodeName string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		podList, err := clientset.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+			FieldSelector: "spec.nodeName=" + nodeName,
+		})
+		if err != nil {
+			klog.Errorf("Get pod list failed, %v", err)
+			returnErr(fmt.Errorf("Get pods list failed"), w, http.StatusInternalServerError)
+			return
+		}
+		klog.V(5).Infof("Got pod list: %v", podList)
+
+		// Successfully get pod list
+		w.Header().Set("content-type", "text/json")
+		data, err := json.Marshal(podList)
+		if err != nil {
+			klog.Errorf("Marshal pod list failed: %v", err.Error())
+			returnErr(fmt.Errorf("Get pod list failed: data transfer to json format failed."), w, http.StatusInternalServerError)
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		n, err := w.Write(data)
+		if err != nil || n != len(data) {
+			klog.Errorf("Write resp for request, expect %d bytes but write %d bytes with error, %v", len(data), n, err)
+		}
```

Comment:
please use `WriteObject()` to return response, you can dive the code here:
https://github.com/openyurtio/openyurt/blob/master/pkg/yurthub/util/util.go#L180-L210